iree_tests/conftest.py

```diff
@@ -375,12 +375,17 @@ def runtest(self):
             raise e
 
     def test_compile(self):
-        proc = subprocess.run(self.compile_args, capture_output=True, cwd=self.test_cwd)
+        compile_env = os.environ.copy()
+        compile_env["SPEC_LOCATION"] = os.getenv("SPEC_LOCATION", default=self.test_cwd)
```

Comment:
The name `SPEC_LOCATION` is specific to one type of file and is also not scoped to just IREE [tests]. If using an environment variable, I'm thinking we could build off of the `IREE_TEST_FILES` name with a name like `IREE_TEST_PATH_EXTENSION`.

Another idea would be to use substitutions like those in https://llvm.org/docs/CommandGuide/lit.html#substitutions. For example, we could look for occurrences of `%s` when loading the config .json file and substitute the config file's directory path in place of it. I think I like the idea of substitutions more than environment variables since that will involve less setup for CI jobs _and_ developers (no need for `SPEC_LOCATION: ${{ github.workspace }}/iree_tests/specs`). That will partly depend on what the reproducer CLI commands are - see my other comment.

Reply:
Ok we can do this too, although it seems more hardcoded for the position of these files. I guess we are not planning on changing the location of these files very often, but anytime we do, we would have to change the code and not a variable. Also, adding the substitution in the command in test_compile and test_run methods works too right?

Reply:
Is it ok to stick with environment variable because I'm also thinking if we want to expand this to the iree run flags. We would need to substitute the flagfile as well. But, we just add the --flagfile={location of flag file} right now, so not sure where substitution would happen (would have to happen in iree). Although we are adding another env variable, I think this might have the biggest scope.

Reply:
I was imagining the substitution would happen here: https://github.com/nod-ai/SHARK-TestSuite/blob/9e9774632d32674e9ffa44b0a2ffc054f6c590ad/iree_tests/conftest.py#L94-L108

but it could also happen here: https://github.com/nod-ai/SHARK-TestSuite/blob/9e9774632d32674e9ffa44b0a2ffc054f6c590ad/iree_tests/conftest.py#L314-L320

I'm trying to focus on the reproducer and what a developer would author for a test, then work backwards from there.

In this case it is `cd /mnt/share/amdsharks/actions-runner/_work/SHARK-TestSuite/SHARK-TestSuite/iree_tests/pytorch/models/sdxl-scheduled-unet-3-tank && iree-run-module --module=model_gpu_rocm_real_weights.vmfb --device=hip --parameters=model=real_weights.irpa --module=sdxl_scheduled_unet_pipeline_fp16_rocm.vmfb --input=1x4x128x128xf16=@inference_input.0.bin --input=2x64x2048xf16=@inference_input.1.bin --input=2x1280xf16=@inference_input.2.bin --input=1xf16=@inference_input.3.bin --expected_output=1x4x128x128xf16=@inference_output.0.bin --expected_f16_threshold=0.8f --flagfile=real_weights_data_flags.txt`

The `cd [directory]` gets files like `@inference_input.1.bin` and `real_weights_data_flags.txt` on the PATH. For the compile command, `--iree-codegen-transform-dialect-library=${SPEC_LOCATION}/attention_and_matmul_spec.mlir` is missing. If we also log `set SPEC_LOCATION=/absolute/path/to/specs/` that should work. With a substitution, we'd instead have `--iree-codegen-transform-dialect-library=/absolute/path/to/specs/attention_and_matmul_spec.mlir`